File: android/app/src/main/java/com/pms_parkin_mobile/api/SimpleCallback.java
package com.pms_parkin_mobile.api;

import android.util.Log;

import retrofit2.Call;
import retrofit2.Callback;

public abstract class SimpleCallback<T> implements Callback<T> {
    private final String tag;

    public SimpleCallback(String tag) {
        this.tag = tag;
    }

    @Override
    public void onFailure(Call<T> call, Throwable t) {
        Log.e(tag, "Network Error: " + t.getMessage());
    }
}

File: android/app/src/main/java/com/pms_parkin_mobile/api/SimpleCallback.kt
package com.pms_parkin_mobile.api

import android.util.Log
import retrofit2.Call
import retrofit2.Callback

abstract class SimpleCallback<T>(private val tag: String?) : Callback<T?> {
    override fun onFailure(call: Call<T?>, t: Throwable) {
        Log.e(tag, "Network Error: " + t.message)
    }
}
